app.R

```diff
@@ -114,17 +115,19 @@ render_trait_plot <- function(subexp_name, input, output, full_cache_data) {
     units <- selected_subexp_data[[ 'trait_data' ]][[ selected_variable ]][[ 'units' ]]
     title <- ifelse(units == '', selected_variable, paste0(selected_variable, ' (', units, ')'))
     
-    trait_plot <- ggplot() + 
-      geom_violin(data = plot_data, scale = 'width', width = 1,
-                   aes(x = as.Date(date), y = mean, 
-                       group = as.Date(date))) +
-      geom_boxplot(data = plot_data, 
-                  aes(x = as.Date(date), y = mean, 
-                      group = as.Date(date)), 
-                  outlier.alpha = 0.25, width = 0.2)  
-#      geom_point(data = plot_data, 
-#                 aes(x = as.Date(date), y = mean), 
-#                 alpha = 0.1, size = 0.1, position = position_jitter(width = 0.1))
+    num_unique_vals = length(unique(plot_data[[ 'mean' ]]))
```

Comment:
`<-` not `=` see https://stackoverflow.com/questions/1741820/assignment-operators-in-r-and